src/code/z_construct.c

```diff
@@ -494,8 +494,8 @@ void Regs_InitDataImpl(void) {
     XREG(93) = 100;
     XREG(94) = 160;
     XREG(95) = 200;
-    WREG(2) = -6080;
-    WREG(3) = 9355;
+    R_PAUSE_OFFSET_VERTICAL = -6080;
+    R_PAUSE_OFFSET_DEPTH = 9355;
```

Comment:
I'm guessing DEPTH is the distance from the camera to the page, but what does VERTICAL mean? Why does it seemingly shift between 0 and -6240?

Reply:
There is a comment on the definition of `R_PAUSE_OFFSET_DEPTH`

`R_PAUSE_OFFSET_VERTICAL` is 0 while the pause menu is open. It gets set to -6240 at the beginning of opening the pause menu and when closing the pause menu.

<details>

<summary>Here are the two consecutive frames on opening where it changes:</summary>

![image](https://github.com/user-attachments/assets/c798f518-f340-4b1c-a98d-54a56c6aa473)
![image](https://github.com/user-attachments/assets/16ffc61d-8672-4292-b418-f4cbb8d70116)

</details>

<details>

<summary>And the two consecutive frames on closing:</summary>

![image](https://github.com/user-attachments/assets/3182e396-92ec-4692-9c33-2b95a732a175)
![image](https://github.com/user-attachments/assets/0e333cfe-2558-46a7-8e8d-1346739408a3)

</details>

I'm not seeing a difference either

<details>

<summary>If I remove the `= 0`, making the value stay at -6240 throughout, here's what it looks like when open:</summary>

![image](https://github.com/user-attachments/assets/ce738bad-baea-481f-99cc-c4b650f59979)

</details>

I'll try dig more to understand what's happening :thinking:

Reply:
Seems like it uses 2 different coordinate systems for drawing when the pause menu is open vs when it is opening/closing? I wonder if the value -6240 can be derived somehow

Reply:
Wow this is jank

On opening and closing the menu, `pauseCtx->offsetY` and `R_PAUSE_OFFSET_DEPTH` are respectively set to `80` and `-6240 = -80 * 100 * 0.78`. Otherwise they're both 0

`pauseCtx->offsetY` is used in the vertices generation, so closest to model space (before pitch rotation is applied), and `R_PAUSE_OFFSET_DEPTH` is applied after rotation (and 0.78 scale).

When these are both 0, the pages rotate around their center (which is only seen when pressing B to save).
When they're set to the other values, they rotate around their lower edge (when opening/closing)

Reply:
I tried to document this c063f56bc640ac1abcc2537601cefbb88e214abd

Reply:
Thanks for digging, makes sense as the place where pages rotate around